arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/call/CallSection.java

```diff
@@ -312,6 +302,75 @@ private void abortingCall(Hub hub) {
     commonValues.collectChildStipend(hub);
   }
 
+  /**
+   * Sets the scenario to the relevant undefined variant, i.e. either
+   *
+   * <p>- {@link
+   * net.consensys.linea.zktracer.module.hub.fragment.scenario.CallScenarioFragment.CallScenario#CALL_PRC_UNDEFINED}
+   *
+   * <p>- {@link
+   * net.consensys.linea.zktracer.module.hub.fragment.scenario.CallScenarioFragment.CallScenario#CALL_SMC_UNDEFINED}
+   *
+   * <p>- {@link
+   * net.consensys.linea.zktracer.module.hub.fragment.scenario.CallScenarioFragment.CallScenario#CALL_EOA_UNDEFINED}
+   *
+   * <p>depending on the address.
+   *
+   * @param hub
+   */
+  private void refineUndefinedScenario(Hub hub) {
+
+    final boolean aborts = hub.pch().abortingConditions().any();
+    if (aborts) {
+      scenarioFragment.setScenario(CALL_ABORT_WONT_REVERT);
+      return;
+    }
+
+    final WorldUpdater world = hub.currentFrame().frame().getWorldUpdater();
+    if (isPrecompile(calleeAddress)) {
+      precompileAddress = Optional.of(calleeAddress);
+      scenarioFragment.setScenario(CALL_PRC_UNDEFINED);
+      // Account rows for precompile are traced at contextReEntry
+
+      precompileSubsection =
```

Comment:
would make more sense to do this in `prcProcessing()`. In this method, we should just set the scenario.

Reply:
Which part ?

Reply:
everythig other than setting the scenario :

precompileAddress = Optional.of(calleeAddress);
         precompileSubsection =  ADDRESS_TO_PRECOMPILE.get(preOpcodeCalleeSnapshot.address()).apply(hub, this);